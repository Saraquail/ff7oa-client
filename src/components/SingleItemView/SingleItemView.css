

.single-monster {
  position: relative;
  left: 25%;
  background-image: linear-gradient(to bottom right, rgba(14,51,95,.7), rgba(4, 3, 44, 0.7));
  border-radius: 1em;
  border: solid 1px #424542;
  box-shadow: 1px 2px #e7dfe7,
              -1px -1px #e7dfe7,
              /* 1px -1px #e7dfe7,
              -1px 1px #e7dfe7,
              -2px 0 #7b757b, */
              0 2px #424542;
  width: 50%;
  color: whitesmoke;
  margin-bottom: 1em; 
}
