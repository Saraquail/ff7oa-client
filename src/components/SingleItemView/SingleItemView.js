import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MonsterApiService from '../../services/monster-api-service';
import './SingleItemView.css';

class SingleItemView extends Component {
  state = {
    mon: {},
  }

  componentDidMount() {
    this.getItem();
  }

  getItem =() => {
    const { monster_id: id } = this.props;

    MonsterApiService.getMonsterById(id)
      .then((data) => this.setState({ mon: data }));
  }

  renderMonster = () => {
    const { mon } = this.state;

    return (
      <div className="single-monster">
        <h2>{mon.name}</h2>
        <p>
          {' '}
          Level:
          {mon.level}
        </p>
        <p>
          {' '}
          HP:
          {mon.hp}
          {' '}
          MP:
          {mon.mp}
        </p>
        <p>
          {' '}
          Exp:
          {mon.exp}
          {' '}
          Ap:
          {mon.ap}
        </p>
        <p>
          {' '}
          Gil:
          {mon.gil}
        </p>
        <p>
          {' '}
          Strength:
          {mon.strength}
        </p>
        <p>
          {' '}
          Weakness:
          {mon.weakness}
        </p>
        <p>
          {' '}
          Location:
          {mon.location}
        </p>
        <p>
          {' '}
          Steal:
          {mon.steal}
        </p>
        <p>
          {' '}
          Drop:
          {mon.drops}
          {' '}
          Morph:
          {mon.morph}
        </p>
        <p>
          {' '}
          Enemy_skill:
          {mon.enemy_skill}
        </p>
      </div>
    );
  }

  render() {
    return (
      <div>
        {this.renderMonster()}
      </div>
    );
  }
}

SingleItemView.propTypes = {
  monster_id: PropTypes.string,
};

export default SingleItemView;
