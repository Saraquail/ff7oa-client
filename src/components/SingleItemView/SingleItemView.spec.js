import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import SingleItemView from './SingleItemView.js';

describe('<SingleItemView/>', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <BrowserRouter>
        <SingleItemView />
      </BrowserRouter>, div,
    );

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <SingleItemView />
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
