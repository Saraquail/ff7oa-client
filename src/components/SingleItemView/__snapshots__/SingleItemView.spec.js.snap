// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleItemView/> renders the UI as expected 1`] = `
<div>
  <div
    className="single-monster"
  >
    <h2 />
    <p>
       
      Level:
    </p>
    <p>
       
      Added By:
    </p>
    <p>
       
      HP:
       
      MP:
    </p>
    <p>
       
      Exp:
       
      Gil:
    </p>
    <p>
       
      Strength:
    </p>
    <p>
       
      Weakness:
    </p>
    <p>
       
      Location:
    </p>
    <p>
       
      Steal:
    </p>
    <p>
       
      Drop:
    </p>
    <p>
       
      Enemy_skill:
    </p>
  </div>
</div>
`;
